### PR TITLE
Feature/tdr 22/access control interface

### PR DESCRIPTION
**Pull request by jbout**
Implement the more generic access control interface as well, allowing us to merge the interfaces

Also prevents editing of ACL if funcAcl is not the current implementation

### CODE REVIEW

**Comment by gitromba** [comment]
- [] New code is covered by tests (if applicable)
- [] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [] New code is respecting code style rules
- [] New code is respecting best practices
- [] New code is not subject to concurrency issues (if applicable)
- [] Feature is working correctly on my local machine (if applicable)
- [] Acceptance criteria are respected
- [] Pull request title and description are meaningful
- [] Changelog is updated according to changes (if applicable)
- [] Documentation is updated according to changes (if applicable)

**Comment by gitromba** [comment]
I cannot login to my local TAO CE instance using this branch. @jbout is this expected?

**Comment by bugalot** [comment]
Me neither I cannot login after a Fresh Install @jbout . However, and I run an Update Process from develop to your branch, it works nice.

**Comment by gitromba** [comment]
The error message, `Access denied. Please renew your authentication!`, pops up immediately once I visit the main page (/). Trying to login with valid credentials results in same error message. 

Here are the respective logs
`[tao]Access to taoCe::oat\taoCe\actions\Main::rootEntry denied for user '' (tao_models_classes_AccessDeniedException,/var/www/html/tao/models/classes/routing/ActionEnforcer.php,167)
[tao]Invoking tao_actions_Main::login by  (GENERIS,CLEARRFW)
[tao]Invoking tao_actions_ClientConfig::config by  (GENERIS,CLEARRFW)
[tao]Invoking tao_actions_Main::login by  (GENERIS,CLEARRFW)
[tao]Unsuccessful login of user 'admin'. ()
[tao]Invoking tao_actions_ClientConfig::config by  (GENERIS,CLEARRFW)`

**Comment by jbout** [comment]
Issue found, we cache the implementation of funcAcl in Tao, so during the install the Access Right Grants do not get applied to funcAcl, will provide a fix

**Comment by jbout** [comment]
Fixed by https://github.com/oat-sa/tao-core/pull/2536 (plan is to drop AclProxy static functions)

**Comment by gitromba** [comment]
A new install with the ` "oat-sa/tao-core": "dev-fix/cached-func-acl",` solves the login issue.

File: manifest.php [hunk 1]
```diff
@@ -20,67 +20,46 @@
...
     'author' => 'Open Assessment Technologies, CRP Henri Tudor',
     'requires' => [
-        'tao' => '>=27.0.0',
-        'generis' => '>=12.15.0'
-    ],
-    'models' => [
-        'http://www.tao.lu/Ontologies/taoFuncACL.rdf'
+        'tao' => '>=44.6.1',
+        'generis' => '>=12.15.0',
+        'taoBackOffice' => '>=4.3.1',
```
[the remark this answers is not in the document]
Comment (jbout) [reply manifest.php:38]: The GUI that allows a user to change the roles requires the BackOffice to be installed

[the remark this answers is not in the document]
Comment (gitromba) [reply manifest.php:38]: It's was a hidden dependency.

**Pull request merged**